### DISCUSSION TITLE
How is this library different from Robot Framework's Browser library?

### DISCUSSION THREAD

**Discussion by wlad** [qahive/robotframework-puppeteer#142] [Q&A] [answered]
There is the [Browser](https://github.com/MarketSquare/robotframework-browser) library which is based on playwright and is developed by the same guys who've build SeleniumLibrary. How is this library different from the Browser library?

**Comment by atthaboon** [comment, marked as answer]
Thanks a lot for you question.

robotframework-puppeteer first we aim to use puppeteer library as our lower layer.
But while waiting for the python library. We enhance the lower layer to able to plug and play with another engine like playwrights.

The Library major goal are:

1. **Flexibility**: create a web test library that has the same interface (keyword) and still allow to use of different lower test library (playwright, puppeteer)

2. **Stable**:  by relying on officials and reducing dependency as much as possible
The lower level implementation for robotframework-puppeteer is different from Brower library. 
robotframework-puppeteer will use the official python library to connect to the playwright directly.

3. **Simple**: as much as possible
Most keyword we try to align with SeleniumLibrary that allow the old script can easily migrate.

**Reply by wlad** [reply]
Thank you. I though playwright is a successor of puppeteer so I was a bit confused. But after reading [this](https://blog.logrocket.com/playwright-vs-puppeteer/) I realise that puppeteer is still an actively maintained project:) 

Is my assumption right, that you want to use https://github.com/pyppeteer/pyppeteer (as puppeteer wrapper) and https://github.com/microsoft/playwright-python (as playwright wrapper)? When importing robotframework-puppeteer the user can choose which one of the two is used under the hood?

If this lib is going to combine both low level layers (puppeteer AND playwright) under the hood then I'm not sure the name robotframework-puppeteer is a good choise.

**Reply by atthaboon** [reply]
Is my assumption right, that you want to use https://github.com/pyppeteer/pyppeteer (as puppeteer wrapper) and https://github.com/microsoft/playwright-python (as playwright wrapper)? When importing robotframework-puppeteer the user can choose which one of the two is used under the hood?
Yes currently, user can choose from browser name 'pychrome' for playwright / 'ptchrome' for puppeteer.

**Reply by atthaboon** [reply]
I'm not sure the name robotframework-puppeteer is a good choice.

Completely agree I still thinking for a while about the new library name. 
My plan will be fork this library to be a new library name after we clear all remaining major issues.
